app/services/subject_mapper.rb

```diff
@@ -134,62 +102,112 @@ def to_s
-      GroupedSubjectMapping.new(ucas_input_subjects, dfe_subject)
+  #  Does the subject list mention languages but hasn't already been covered?
+  class MFLOtherMapping
+    def applicable_to?(ucas_subjects)
+      (ucas_subjects & language_categories).any? &&
+        (ucas_subjects & mandarin).none? &&
+        (ucas_subjects & mfl_main).none?
     end
 
-    secondary_subjects = []
+    def to_s
+      "Modern languages (other)"
+    end
 
-    secondary_subjects += secondary_subject_mappings.map { |mapping|
-      mapping.to_s if mapping.applicable_to?(ucas_subjects)
-    }.compact
+  private
 
-    ucas_language_cat = ["languages",
-                         "languages (african)",
-                         "languages (asian)",
-                         "languages (european)"]
+    def language_categories
+      ["languages", "languages (african)", "languages (asian)", "languages (european)"]
+    end
 
-    ucas_mfl_mandarin = %w[chinese mandarin]
+    def mandarin
+      %w[chinese mandarin]
+    end
 
-    ucas_mfl_main = ["english as a second or other language",
-                     "french",
-                     "german",
-                     "italian",
-                     "japanese",
-                     "russian",
-                     "spanish"]
+    def mfl_main
+      ["english as a second or other language",
+       "french",
+       "german",
+       "italian",
+       "japanese",
+       "russian",
+       "spanish"]
+    end
+  end
 
-    #  Does the subject list mention languages but hasn't already been covered?
-    if (ucas_subjects & ucas_language_cat).any? && (ucas_subjects & ucas_mfl_mandarin).none? && (ucas_subjects & ucas_mfl_main).none?
-      secondary_subjects.push("Modern languages (other)")
+  # TODO: remove this bonkers logic once course mapping is done by one app!
+  # The user need for this is unclear
+  #
+  # Does the subject list mention english, and it's mentioned in the title (or it's the only subject we know for this course)?
+  class SecondaryEnglishMapping
+    def initialize(course_title)
+      @course_title = course_title
     end
 
-    # Does the subject list mention a subject we are happy to translate if the course title contains a mention?
-    (ucas_subjects & @ucas_needs_mention_in_title.keys).each do |ucas_subject|
-      if course_title.match?(@ucas_needs_mention_in_title[ucas_subject])
-        secondary_subjects.push(map_to_subject_name(ucas_subject))
-      end
+    def applicable_to?(ucas_subjects)
+      (ucas_subjects & ucas_english).any? &&
+        @course_title.index("english") != nil
     end
 
-    # Does the subject list mention english, and it's mentioned in the title (or it's the only subject we know for this course)?
-    if (ucas_subjects & @ucas_english).any?
-      if secondary_subjects.none? || course_title.index("english") != nil
-        secondary_subjects.push("English")
-      end
+    def to_s
+      "English"
+    end
+
+  private
+
+    def ucas_english
```

Comment:
Can we shove that in  the `Mapping` and remove SecondaryEnglishMapping class altogether?

Reply:
unfortunately not yet – some of the prod data specs fail when I tried it, I believe. I'd prefer to do this simplification on both the C# and the Rails side in a short while.

Reply:
It's is a deeper rabbit hole, another time, 

Basically `English` mapping relies on other mapping to firstly before attempting to map `English`

Reply:
Any reason not to make this a constant?